### ISSUE TITLE
Consider placement of `brms` reparameterisation code

### ISSUE THREAD

**Issue by athowes** [epinowcast/epidist#276]
In https://github.com/epinowcast/epidist/pull/262 we added a reparameterisation of the gamma distribution so that the input into Stan matches `brms` approach. It mostly amounts to:

```r
if (family_name == "gamma") {
  lpdf_dpars <- c("shape", "shape ./ mu")
}
```

This is currently housed in `epidist_stancode`. However, perhaps it'd be best to move it into `epidist_family` given that it is family related. Then each `epidist_family.latent_family` would have its own reparameterisation if required. The only hitch here is then how to store this. It does seem like one could just extent `brms` family objects with e.g. `family$reparam <- "..."`.

More generally, this is still a bit of a patch solution, but I think it might be favoured over deeper integration with `brms` approach to reparameterisation (which seems quite complicated, see https://github.com/epinowcast/epidist/issues/226).

**Comment by seabbs** [comment]
IMO this needs to be done very much sooner rather than later as it will really add to our technical debt. We also have a working solution

**Comment by athowes** [comment]
I'm happy to do this sooner rather than later. I don't have a good grasp on why in particular this would in particular ("really") add to our technical debt given that 1. people are not likely to use many families, 2. for (I think) most families the `brms` and Stan parameterisations are the same. But anyway, as we have a solution happy to move it forward.

**Comment by seabbs** [comment]
because its embedded in a model family and we very shortly want to start adding new models imo

**Comment by athowes** [comment]
Note that `epidist_family.latent_family` is model-family specific but this reparameterisation is family specific i.e. we might end up with duplicated reparameterisations across models.

**Comment by athowes** [comment]
I've looked and this is harder than I remembered because:

```r
epidist_family.epidist_latent_individual
```

i.e. `epidist_family` is dispatching on model not family. So I could move it into `epidist_family` but then we basically have the same problem.

I think this is suggesting there could be something which is separate to "families" which might be shared across models. Then maybe that can be called from within `epidist_family.epidist_latent_individual`.
